modules/tensor_mechanics/src/userobjects/CrystalPlasticityStateVariable.C

```diff
@@ -42,24 +46,58 @@ CrystalPlasticityStateVariable::CrystalPlasticityStateVariable(const InputParame
 void
 CrystalPlasticityStateVariable::initSlipSysProps(std::vector<Real> & val) const
 {
-  readFileInitSlipSysRes(val);
+  switch (_intvar_read_type)
```

Comment:
This is just my preference, but I would put in two `case` statements (one for each valid option) and have the `default` error out. This makes sure the won't be any surprises when someone modifies the moose enum above and forgets the switch, or if there is an illegal memory access that overwrites the variable etc.